Some issues building and compiling .erl

Hi!

This is a great project! I've tried following the build steps on MacOS but got this error:

```
/Users/erszcz/work/lumen/lumen/compiler/codegen/lib/lumen/compiler/Translation/MLIR.cpp:37:27: error: no matching constructor for initialization of 'mlir::LLVM::LLVMDialect'
  auto *llvmDialect = new mlir::LLVM::LLVMDialect(mlirContext, llvmContext);
                          ^                       ~~~~~~~~~~~~~~~~~~~~~~~~
/Users/erszcz/.local/share/llvm/lumen/include/mlir/Dialect/LLVMIR/LLVMOpsDialect.h.inc:12:12: note: candidate constructor not viable: requires single argument 'context', but 2 arguments were provided
  explicit LLVMDialect(::mlir::MLIRContext *context);
           ^
/Users/erszcz/.local/share/llvm/lumen/include/mlir/Dialect/LLVMIR/LLVMOpsDialect.h.inc:10:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided
class LLVMDialect : public ::mlir::Dialect {
      ^
1 error generated.
```

Environment details:

- commit 8cfd9dfe, i.e. current `develop`

- Rust toolchain

```
$ rustup --version
rustup 1.21.1 (7832b2ebe 2019-12-20)
$ cargo --version
cargo 1.45.0-nightly (cb06cb269 2020-05-08)
$ rustc --version
rustc 1.45.0-nightly (769d12eec 2020-05-12)
```
- LLVM installed by following https://github.com/lumen/lumen/tree/8cfd9dfeeecca1c29da9fab3a4feafeeae4d9624#macos

However, applying this patch allows me to build the project:

```
$ git diff
diff --git a/compiler/codegen/lib/lumen/compiler/Translation/MLIR.cpp b/compiler/codegen/lib/lumen/compiler/Translation/MLIR.cpp
index ce6dc9cf..a0316bc5 100644
--- a/compiler/codegen/lib/lumen/compiler/Translation/MLIR.cpp
+++ b/compiler/codegen/lib/lumen/compiler/Translation/MLIR.cpp
@@ -34,7 +34,7 @@ extern "C" void MLIRRegisterDialects(MLIRContextRef mlirCtx, LLVMContextRef llvm
   // NOTE: The dialect constructors internally call registerDialect,
   // which moves ownership of the dialect objects to the MLIRContext,
   // so we don't have to manage them ourselves.
-  auto *llvmDialect = new mlir::LLVM::LLVMDialect(mlirContext, llvmContext);
+  auto *llvmDialect = new mlir::LLVM::LLVMDialect(mlirContext);
   auto *eirDialect = new lumen::eir::EirDialect(mlirContext);
 }
```

Alas, it's not enough to get the compiler to work:

```
$ cat hello.erl
-module(test).
-compile(export_all).

main(_) ->
    io:format("hello\n", []).
$ ./bin/lumen compile -lc hello.erl
... debug output elided
Assertion failed: (ScopedContext::getCurrentScopedContext() && "Unexpected Null ScopedContext"), function getBuilder, file /Users/paulschoenfelder/src/github.com/bitwalker/llvm-project/mlir/lib/EDSC/Builders.cpp, line 54.
Abort trap: 6
```

Just thinking out loud, but might it be that the README on `develop` is not up to date with regard to LLVM version to use?

--- a/.github/workflows/x86_64-apple-darwin-libraries.yml
+++ b/.github/workflows/x86_64-apple-darwin-libraries.yml
@@ -17,7 +17,7 @@ jobs:
         run: |
           mkdir -p ~/.local/share/llvm
           pushd ~/.local/share/llvm
-          wget https://github.com/lumen/llvm-project/releases/download/lumen-10.0.0-dev_2020-04-16/clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
+          wget https://github.com/lumen/llvm-project/releases/download/lumen-10.0.0-dev_2020-04-26/clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
           tar xvfz clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
           mv clang+llvm-10.0.0-x86_64-apple-darwin19.3.0 lumen
           popd

--- a/.github/workflows/x86_64-apple-darwin-runtime_full.yml
+++ b/.github/workflows/x86_64-apple-darwin-runtime_full.yml
@@ -17,7 +17,7 @@ jobs:
         run: |
           mkdir -p ~/.local/share/llvm
           pushd ~/.local/share/llvm
-          wget https://github.com/lumen/llvm-project/releases/download/lumen-10.0.0-dev_2020-04-16/clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
+          wget https://github.com/lumen/llvm-project/releases/download/lumen-10.0.0-dev_2020-04-26/clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
           tar xvfz clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
           mv clang+llvm-10.0.0-x86_64-apple-darwin19.3.0 lumen
           popd

--- a/README.md
+++ b/README.md
@@ -72,7 +72,7 @@ the usual default for this XDG variable.
 
     mkdir -p $XDG_DATA_HOME/llvm/lumen
     cd $XDG_DATA_HOME/llvm/lumen
-    wget https://github.com/lumen/llvm-project/releases/download/lumen-10.0.0-dev_2020-04-16/clang+llvm-10.0.0-x86_64-linux-gnu.tar.gz
+    https://github.com/lumen/llvm-project/releases/download/lumen-10.0.0-dev_2020-04-26/clang+llvm-10.0.0-x86_64-linux-gnu.tar.gz
     tar -xz --strip-components 1 -f clang+llvm-10.0.0-x86_64-linux-gnu.tar.gz
     rm clang+llvm-10.0.0-x86_64-linux-gnu.tar.gz
     cd -
@@ -81,7 +81,7 @@ the usual default for this XDG variable.
 
     mkdir -p $XDG_DATA_HOME/llvm
     cd $XDG_DATA_HOME/llvm/lumen
-    wget https://github.com/lumen/llvm-project/releases/download/lumen-10.0.0-dev_2020-04-16/clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
+    wget https://github.com/lumen/llvm-project/releases/download/lumen-10.0.0-dev_2020-04-26/clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
     tar -xzf clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
     rm clang+llvm-10.0.0-x86_64-apple-darwin19.3.0.tar.gz
     mv clang+llvm-10.0.0-x86_64-apple-darwin19.3.0 lumen

